### DISCUSSION TITLE
Angular nginx oidc client - microsoft azure api token call from app creates CORS issue

### DISCUSSION THREAD

**Discussion by Sethujag** [community/community#60623] [New to GitHub]
Hi everyone, hope you are all doing good.
We are having issue one of our application which is built with Angular 8.0 and using oidc client. Token based 0auth2 was initially built with ADFS (Active Directory Federation Services (AD FS) is a feature of the Windows Server operating system that extends end users' single sign-on (SSO) access to applications and systems outside the corporate firewall.). 
Now as part of a project - migrating that to Azure Active Directory auth process. Did the necessary design changes point Azure AD (AAD) instead ADFS, which is working perfectly in local (localhost). But when we try to containerize and deploy in the server, it is failing with CORS Policy error.

**Error:**
Access to XMLHttpRequest at 'https://login.microsoftonline.com/<<client-id/tenent-id>>/oauth2/v2.0/token' (redirected from 'https://<<applicaton-uri>>/token') from origin ''https://<<applicaton-uri>>' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

The token call is configured in the nginx.conf file which has following details like:

    location /token {
      proxy_pass https://login.microsoftonline.com/<<client-id/tenent-id>>/oauth2/v2.0/token;
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;      
      proxy_set_header Connection 'upgrade';
      proxy_set_header Host $host;
    }

I tried adding the below parms based on some inputs I found online, but not worked out. Alternatively I also tried adding add_header instead proxy_set_header that also not worked. 
      proxy_set_header Access-Control-Allow-Origin '*';
      proxy_cache_bypass $http_upgrade;
      proxy_cookie_path / "/; HTTPOnly; Secure";

Also tried most of the option not and didn't get the expected result. 

Just one question here is - how to pass the allow-access-control-origin header with the request header. Any expertise inputs will be much appreciated and will help us to resolve the issue.

**Comment by queenofcorgis** [comment]
Hi @Sethujag ,

Thanks for being a part of the GitHub Community, we're glad you're here!

If you're looking for help for this specific topic, you might want to try asking for help somewhere that focuses on Azure like their community: https://techcommunity.microsoft.com/t5/azure/ct-p/Azure. It's possible that another GitHub user might have run into this same issue and can help, but the GitHub Community on Discussions focuses primarily on topics related to GitHub itself or collaboration on project development and ideas. We want to make sure you’re getting the best support you can, but this space may not be the right place for this particular topic.

Best of luck!
